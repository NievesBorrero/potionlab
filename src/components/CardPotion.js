import React from 'react'
import { connect } from 'react-redux'
import { PropTypes } from 'prop-types'

import { Card } from './Card'
import { addPotion } from '../actions/chest'
import { ImageContainer } from './containers/ImageContainer'
import { Title } from './Text/Title'
import { primaryColor } from '../constants/colors'
import { Text } from './Text/Text'
import { ButtonCardContainer } from './containers/ButtonCardContainer'
import { Button } from './Button'

export const CardPotion = props => {
  const { potion, addPotion } = props

  const onClick = e => {
    e.preventDefault()
    addPotion(potion)
  }

  return (
    <Card>
      <ImageContainer>
        <img width="100%" src={potion.image_thumb} />
      </ImageContainer>
      <Title color={primaryColor}>
        {potion.prize} {potion.currency}
      </Title>
      <Text>{potion.name}</Text>
      <ButtonCardContainer>
        <Button secondary onClick={e => onClick(e)}>AÃ±adir al carrito</Button>
      </ButtonCardContainer>
    </Card>
  )
}

CardPotion.propTypes = {
  potion: PropTypes.object,
  addPotion: PropTypes.func
}

function mapStateToProps() {
  return {}
}

export default connect(mapStateToProps, { addPotion })(CardPotion)
