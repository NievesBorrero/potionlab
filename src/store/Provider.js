import React from 'react'
import { Provider } from 'react-redux'

const ProviderWrapper = ({ children, store }) => (
  <Provider store={store}>
    { children }
  </Provider>
)

export default ProviderWrapper
