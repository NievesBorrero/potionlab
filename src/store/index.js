import Raven from 'raven-js'
import { createStore, applyMiddleware, compose } from 'redux'
import createSagaMiddleware from 'redux-saga'
import { createHashHistory } from 'history'
import { routerMiddleware } from 'connected-react-router'
import createRavenMiddleware from 'raven-for-redux'

import rootReducer from '../reducers/index'
import rootSaga from '../sagas'

Raven.config(process.env.SENTRY_DSN_FRONTEND, { tags: ['frontend'] }).install()

export const history = createHashHistory()
const sagaMiddleware = createSagaMiddleware()
const storeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

export default function configureStore (preloadedState) {
  const store = createStore(
    rootReducer(history),
    preloadedState,
    storeEnhancers(
      applyMiddleware(
        routerMiddleware(history),
        sagaMiddleware,
        createRavenMiddleware(Raven, {})
      )
    )
  )

  sagaMiddleware.run(rootSaga)

  return store
}
