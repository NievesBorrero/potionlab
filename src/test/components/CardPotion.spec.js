import React from 'react'
import { shallow } from 'enzyme'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { ConnectedRouter, routerMiddleware } from 'connected-react-router'

import { history } from '../../store/index'
import CardPotion from '../../components/CardPotion'

const initialState = {}

const middlewares = [routerMiddleware]
const mockStore = configureStore(middlewares)

describe('CardPotion', () => {
  it('should render correctly in "debug" mode', () => {
    const store = mockStore(initialState)
    const component = shallow(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <CardPotion debug />
        </ConnectedRouter>
      </Provider>
    )

    expect(component).toMatchSnapshot()
  })

  it('should render correctly a potion in the list', () => {
    const store = mockStore(initialState)
    const component = shallow(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <CardPotion debug potion={{name: 'Veneno'}}/>
        </ConnectedRouter>
      </Provider>
    )
    expect(component.html()).toContain('Veneno')
  })
})
