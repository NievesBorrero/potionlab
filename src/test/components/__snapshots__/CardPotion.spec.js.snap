// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardPotion should render correctly in "debug" mode 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {},
    }
  }
>
  <Connect(ConnectedRouterWithContext)
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(CardPotion)
      debug={true}
    />
  </Connect(ConnectedRouterWithContext)>
</ContextProvider>
`;
